sdk/core/Azure.Core/tests/TransportFunctionalTests.cs

```diff
@@ -113,7 +112,7 @@ public async Task CanSetContentLengthOverMaxInt()
                 // Sending the request would fail because of length mismatch
             }
 
-            Assert.AreEqual(requestContentLength, requestContentLength);
+            Assert.AreEqual(long.MaxValue, contentLength);
```

Comment:
Could we add some comments to this test that explain why this test does what we expect and why this assert should be true? As is this seems to be testing some implementation detail of the TestServer.

Reply:
```suggestion
            Assert.AreEqual(infiniteStream.Length, contentLength);
```

Reply:
I'd like to explicitly compare to `long.MaxValue` so we are not reliant on changes in `infiniteStream`

Reply:
But isn't this already reliant on changes to infiniteStream? It will fail if its length ever changes.

Reply:
Sure, but imagine someone changing the `InfiniteStream` to 10000 length. Test will keep passing but not testing what intended.

Reply:
then add an Assert that infiniteStream.Lenth > int.MaxValue?

Reply:
`Assert.AreEqual(long.MaxValue, contentLength);` does exactly that :)

Reply:
![](https://media1.giphy.com/media/3oriOaivTEk4PotVEQ/giphy.gif)

Reply:
Fine, fine...